sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -20,6 +22,35 @@ export class ApplicationOfferingChangeRequestService {
     private readonly applicationOfferingChangeRequestRepo: Repository<ApplicationOfferingChangeRequest>,
   ) {}
 
+  /**
+   * Gets an eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param options method options:
+   * - `applicationId` specific eligible application to be retrieved.
+   * @returns eligible application.
+   */
+  async getEligibleApplications(
+    locationId: number,
+    options: {
+      applicationId?: number;
```

Comment:
Shouldn't we make the applicationId to be mandatory here? like this. 

![image](https://github.com/bcgov/SIMS/assets/54600590/31c04458-ae91-4d54-a265-4e11ca6c97cd)

and same for the pagination?

Reply:
Changed.